### ISSUE TITLE
Node Taints

### ISSUE THREAD

**Issue by coffeepac** [samsung-cnct/k2#298]
We should be able to apply two forms of taints to nodes at node startup in two ways:
- ~~default nodePool~~
- arbitrary config driven taints

taints are the future for how pods will be scheduled to nodes and as of kubernetes v1.6 they can be applied to nodes via kubelet CLI options.  we should enable this functionality in k2

**Comment by coffeepac** [comment]
this is blocked by our ability to deploy kubernetes v1.6.0.  this is currently tracked in #283 and #184  There may be more issues but this is all we've found so far.

**Comment by DStorck** [comment]
What are the deliverables for this ticket ?

**Comment by davidewatson** [comment]
There are (at least) two uses for taints:

1) A user facing option to apply taints to node pools, a la https://github.com/samsung-cnct/k2/pull/452/files

2) Removing `--register-schedulable=false` from kubelet on the master nodes and replacing it with taints.

**Comment by venezia** [comment]
`--register-scheduable=false` is now deprecated in lieu of taints - so it is an important cleanup task

**Comment by DStorck** [comment]
everything i'm finding seems to indicate that we set taints with kubectl, how does this integrate to cluster up?

**Comment by venezia** [comment]
https://kubernetes.io/docs/admin/kubelet/ - --register-with-taints - was released in 1.6

**Comment by coffeepac** [comment]
important note:  because this change relies on a feature that is not backwards compatible you will need to version the affected templates.  the dns templates provide an example of how to do this to maintain backwards compatibility.

**Comment by DStorck** [comment]
@coffeepac I think there is already a v1.6 directory, I was planning to just edit the template in there https://github.com/samsung-cnct/k2/tree/master/ansible/roles/kraken.nodePool/kraken.nodePool.master/templates

**Comment by coffeepac** [comment]
stellar!  I wasn't sure.  we don't have backwards compatibility CI tests so I like to be proactive

**Comment by DStorck** [comment]
work to restrict master nodes using t & t is done here: https://github.com/samsung-cnct/k2/pull/588

**Comment by DStorck** [comment]
user-facing option to add taints to nodes here: https://github.com/samsung-cnct/k2/pull/610

**Comment by DStorck** [comment]
Both PRs were merged.  A new issue was made to address how GKE will add a user-facing option for taints. https://github.com/samsung-cnct/k2/issues/631
